test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/MutableSubscriptionSetTests.kt

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.test.mongodb.shared
+
+import io.realm.Realm
+import io.realm.entities.sync.flx.FlexChildObject
+import io.realm.entities.sync.flx.FlexParentObject
+import io.realm.internal.platform.runBlocking
+import io.realm.mongodb.subscriptions
+import io.realm.mongodb.sync.Subscription
+import io.realm.mongodb.sync.SubscriptionSetState
+import io.realm.mongodb.sync.SyncConfiguration
+import io.realm.query
+import io.realm.test.mongodb.TEST_APP_FLEX
+import io.realm.test.mongodb.TestApp
+import io.realm.test.mongodb.createUserAndLogIn
+import io.realm.test.util.TestHelper
+import io.realm.test.util.toRealmInstant
+import kotlinx.coroutines.delay
+import kotlinx.datetime.Clock
+import kotlin.RuntimeException
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Ignore
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+/**
+ * Class wrapping tests for modifying a subscription set.
+ */
+class MutableSubscriptionSetTests {
+
+    private lateinit var app: TestApp
+    private lateinit var realm: Realm
+    private lateinit var config: SyncConfiguration
+
+    @BeforeTest
+    fun setup() {
+        app = TestApp(appName = TEST_APP_FLEX)
+        val (email, password) = TestHelper.randomEmail() to "password1234"
+        val user = runBlocking {
+            app.createUserAndLogIn(email, password)
+        }
+        config = SyncConfiguration.Builder(
+            user,
+            schema = setOf(FlexParentObject::class, FlexChildObject::class)
+        )
+            .build()
+        realm = Realm.open(config)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        realm.close()
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun initialSubscriptions() = runBlocking {
+        realm.subscriptions.update {
+            assertEquals(0, size)
+            assertEquals(SubscriptionSetState.UNCOMMITTED, state)
+        }
+        Unit
+    }
+
+    @Test
+    fun addNamedSubscription() = runBlocking {
+        val now = Clock.System.now().toRealmInstant()
+        // on macOS Core and Kotlin apparently doesn't agree on the exact timing, sometimes
```

Comment:
This sound quite critically. As an absolute minimum we should create an issue to investigate.

Reply:
Maybe it is just our `toRealmInstant` that is wrong 🙈

Reply:
Maybe wrap it in a method, so that we don't have to fix it allover when root cause is settled.